### PR TITLE
Contributing guidelines

### PR DESCRIPTION
**Pull request by CerineBnsd**
## Description
Adding contributing guidelines to the project, and set templates for Pull Requests and the different types of issues:
- Improvement
- New Feature
- Refactor
- Documentation
- Testing

The issues will be automatically labeled with the relevant type label.

[**Work still in progress**]

- Contributing guidelines inspired by [Atom Contributing Guidelines](https://github.com/atom/atom/blob/master/CONTRIBUTING.md)
- Labels and issues management inspired from OpenMined. 
## Related Issue
None.

## Type of change

- [X] This change only concerns the documentation

## How Has This Been Tested?
- Test the creation of the Issues from the specified template.
- Test the creation of Pull requests from the template.

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have updated the documentation according to my changes

<!-- 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
-- >

### CODE REVIEW

**Comment by CerineBnsd** [comment]
@JoFrost I am still finishing the writing of the CONTRIBUTING.ms file. However, I'd like to have your opinion regarding the structure of the document, the grouping of labels and the issue template when possible.

**Comment by JoFrost** [comment]
I really like your suggestions and the way you structured the contributing file. I checked the issue template as well and for me this is the good way to go! Nice work! :)

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+Please describe in detail how you tested your changes. Include details of your testing environment, and the tests you ran to see how your change affects other areas of the code, etc.
+
+## Checklist:
+
+- [ ] My code follows the style guidelines of this project
+- [ ] I have performed a self-review of my code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] My changes generate no new warnings
+- [ ] I have updated the documentation according to my changes
```
Comment (JoFrost) [inline .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:30]: Thanks for pointing out this! :)

**Review by JoFrost — COMMENTED** [review]
We will merge as soon as it's ready.

**Comment by CerineBnsd** [comment]
> We will merge as soon as it's ready.

Thank you for the feedback! I'll mention you again once it's done.

**Comment by CerineBnsd** [comment]
If the PR in approved, we'll need to add the labels and configure the templates.

**Pull request merged**